python/dgl/contrib/cugraph/cugraph_storage.py

```diff
@@ -0,0 +1,220 @@
+             If the graph is homogeneous, directly pass the above formats.
+             Otherwise, the argument must be a dictionary with keys being
+             node types and values being the node IDs in the above formats.
+        relabel_nodes : bool, optional
+            If True, the extracted subgraph will only have the nodes in the
+            specified node set and it will relabel the nodes in order.
+        output_device : Framework-specific device context object, optional
+            The output device.  Default is the same as the input graph.
+        Returns
+        -------
+        DGLGraph
+            The subgraph.
+        """
+        sampled_cugraph = self.graphstore.node_subgraph(nodes)
+        # the return type is cugraph subgraph
+        sample_graph = cugraphToDGL(sampled_cugraph)
+        sample_graph.to_device(output_device)
+        return sample_graph
+
+    # Required in Link Prediction
+    # relabel = F we use dgl functions,
+    # relabel = T, we need delete codes and relabel
+    def edge_subgraph(self, edges, relabel_nodes=False, output_device=None):
+        """
+        Return a subgraph induced on given edges.
+        This has the same semantics as ``dgl.edge_subgraph``.
+        Parameters
+        ----------
+        edges : edges or dict[(str, str, str), edges]
+            The edges to form the subgraph. The allowed edges formats are:
+            * Int Tensor: Each element is an edge ID. The tensor must have the
+              same device type and ID data type as the graph's.
+            * iterable[int]: Each element is an edge ID.
+            * Bool Tensor: Each :math:`i^{th}` element is a bool flag
+             indicating whether edge :math:`i` is in the subgraph.
+            If the graph is homogeneous, one can directly pass the above
+            formats. Otherwise, the argument must be a dictionary with keys
+            being edge types and values being the edge IDs in the above formats
+        relabel_nodes : bool, optional
+            If True, the extracted subgraph will only have the nodes in the
+            specified node set and it will relabel the nodes in order.
+        output_device : Framework-specific device context object, optional
+            The output device.  Default is the same as the input graph.
+        Returns
+        -------
+        DGLGraph
+            The subgraph.
+        """
+        pass
+
+    # Required in Link Prediction negative sampler
+    def find_edges(self, edges, etype=None, output_device=None):
+        """Return the source and destination node IDs given the edge IDs within
+         the given edge type.
+         return type is tensor need to change.
+        """
+        # edges are a range of edge IDs, for example 0-100
+        selected_edges = (
+            self._edata[self._edata['_TYPE_'] == etype].iloc[edges])
+        src_nodes = selected_edges['_SRC_']
+        dst_nodes = selected_edges['_DST_']
```

Comment:
The constants for these col names are `PropertyGraph.src_col_name` and `PropertyGraph.dst_col_name`